edx-solutions/api-integration
# WIP [MCKIN-8112] UserList API: allow multiple usernames in filter

This change allows us to query for a subset of users enrolled in a given course.

**JIRA tickets**: MCKIN-8112, OC-4961

**Testing instructions**:

1. Run edx-solutions LMS with this version of api-integration
2. Fetch the user list using both the `courses` and `username` filter options, and your configured `settings.EDX_API_KEY`, e.g.
    ```bash
    curl -X GET \
           --header "Content-Type: application/json" \
           --header "X-Edx-Api-Key: edx_api_key" \
          "http://lms.mcka.local/api/server/users/?courses=edX/DemoX/demo&usernames=honor,audit"
   ```
   Note that the returned data covers only those username(s) enrolled in the given course(s).

To run automated tests, install this branch and run from the edxapp shell:
```bash
NO_PREREQ_INSTALL=1 paver test_system -s lms -t ../../../src/api-integration/edx_solutions_api_integration/
```

**Reviewers**
- [ ] 2nd reviewer TBD
